lib/internal/http2/core.js

```diff
@@ -1703,6 +1705,16 @@ class ClientHttp2Session extends Http2Session {
     if (options.waitForTrailers)
       stream[kState].flags |= STREAM_FLAGS_HAS_TRAILERS;
 
+    const { signal } = options;
+    if (signal) {
```

Comment:
What happens if this is aborted already?

Reply:
That's an excellent question. I don't know what's the expected behavior in that case.

Looking at the DOM implementation, it appears like the expected behavior is to abort immediately. I shall add this (and a test) after my :pizza: :D

Reply:
Yes, it would need to abort immediately.

Reply:
@mcollina Done, PTALA.

Also @benjamingr this is likely something we'd want to do in `http_client` as well.